src/plugins/discover/public/application/main/components/top_nav/get_top_nav_links.tsx

```diff
@@ -163,13 +163,28 @@ export const getTopNavLinks = ({
       if (!services.share || !updatedDataView) {
         return;
       }
-      const sharingData = await getSharingData(searchSource, state.appState.getState(), services);
+
+      const appState = state.appState.getState();
+      const { timefilter } = services.data.query.timefilter;
+      const timeRange = timefilter.getTime();
+      const refreshInterval = timefilter.getRefreshInterval();
+      const { grid, ...otherState } = appState;
+      const relativeUrl = services.locator.getRedirectUrl({
+        ...otherState,
+        ...(dataView?.isPersisted()
+          ? { dataViewId: dataView?.id }
+          : { dataViewSpec: dataView?.toSpec() }),
+        timeRange,
+        refreshInterval,
+      });
+      const shareableUrl = relativeToAbsolute(relativeUrl);
```

Comment:
I think, `getUrl`, with `absolute` set to `true` should do this job
```suggestion
       const shareableUrl = await services.locator.getUrl(
        {
          ...otherState,
          ...(dataView?.isPersisted()
            ? { dataViewId: dataView?.id }
            : { dataViewSpec: dataView?.toSpec() }),
          timeRange,
          refreshInterval,
        },
        { absolute: true }
      );
```

Reply:
Additionally I think you need to consider when a saved search with id is actually loaded

Reply:
Unfortunately I don't think we can use `getUrl` since the ad-hoc data view isn't stored in either global or app state. The redirect service allows us to inject internal state, which is where the ad-hoc state lives.

Reply:
I think we can, have a look in Discover's locator code:

https://github.com/elastic/kibana/blob/503b466b72865433cfa0148409e982aa39bc2f40/src/plugins/discover/common/locator.ts#L178

This is then available in the history's location state

https://github.com/elastic/kibana/blob/668fe89e82db05fedb9a7b8c0b3bd4a61f91909b/src/plugins/discover/public/application/main/discover_main_route.tsx#L127

And we use it to construct a temporary/ad-hoc data view

Reply:
tested, and it works (in your PR, using `getUrl`, create a link, paste it in another browser, temporary data view was created)

Reply:
We chatted about this, and it sounds like we still do need to use the redirect URL. I've also updated it so it should work with saved searches as well.

Reply:
thx Lukas, I was wrong totally on the wrong track here 😅, sry for the confusion